security/kubernetes-nodes.md

```diff
@@ -0,0 +1,280 @@
+We will apply two policies: one for the master nodes. and one for the worker nodes.
+
+> Note: Run this tutorial only on a sandbox cluster; using it on a real cluster can disrupt traffic.
+{: .alert .alert-danger }
+
+> Note: This tutorial was tested on a cluster created with kubeadm v1.18.2 on AWS, using a "stacked etcd" [topology](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/ha-topology/). Stacked etcd topology means the etcd pods are running on the masters. kubeadm uses stacked etcd by default.
+>
+> If your Kubernetes cluster is on a different platform, is running a variant of Kubernetes, or is running a topology with an external etcd cluster,
+> please review the required ports for master and worker nodes in your cluster and adjust the policies in this tutorial as needed.
+{: .alert .alert-info }
+
+First, let's restrict ingress traffic to the master nodes. The ingress policy below contains three rules.
+The first rule allows access to the API server port from anywhere. The second rule allows access to the Kubernetes
+control plane from localhost. These control plane processes includes the etcd server client API, the scheduler, and the controller-manager. This rule
+also whitelists localhost access to the kubelet API and calico/node health checks. And the final rule allows the etcd pods
+to peer with each other.
+
+If you have not modified the failsafe ports, you should still have SSH access to the nodes after applying this policy.
+Now apply the ingress policy for the Kubernetes masters:
+
+```
+calicoctl apply -f - << EOF
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: ingress-k8s-masters
+spec:
+  selector: has(node-role.kubernetes.io/master)
+  # This rule allows ingress to the Kubernetes API server.
+  ingress:
+  - action: Allow
+    protocol: TCP
+    destination:
+      ports:
+      # kube API server
+      - 6443
+  # This rule allows traffic to Kubernetes control plane ports
+  # from localhost. The health check port for calico/node is also whitelisted
+  - action: Allow
+    protocol: TCP
+    destination:
+      nets:
+      - 127.0.0.1/32
+      ports:
+      # etcd server client API
+      - "2379:2381"
+      # kube-scheduler
+      - 10251
+      - 10259
+      # kube-controller-manager
+      - 10252
+      - 10257
+      # kubelet API
+      - 10250
+      # calico/node health check
+      - 9099
+  # This rule is required in multi-master clusters where etcd pods are colocated with the masters.
+  # Allow the etcd pods on the masters to communicate with each other. 2380 is the etcd peer port.
+  - action: Allow
+    protocol: TCP
+    source:
+      selector: has(node-role.kubernetes.io/master)
+    destination:
+      ports:
+      - 2380
+EOF
+```
+
+Note that the above policy selects the standard Kubernetes label **node-role.kubernetes.io/master** attached to master nodes.
```

Comment:
This label is standard in kubeadm clusters.  It may be set on other installers too, but I don't think it is standard across all clusters.  e.g. Rancher clusters don't set this label I believe.  So suggest tweaking this sentence to something like "the **node-role.kubernetes.io/master** label that kubeadm sets on master nodes".

Reply:
3bf0e487e916606982d1129af7b0993237934b52